### PR TITLE
Updated the name of the Azure Resource Group deployment to be unique

### PR DESCRIPTION
**Pull request by antmeehan**
**DeployAzureResourceGroup task**

The deployment is created using the name of the resource group, and hence is overwritten every time. (Not good for lookup up history in the portal)

This change modifies the name of the deployment to be based on the deployment filename and a UTC timestamp (similar to the Azure Resource Group project is Visual Studio).

### CODE REVIEW

File: Tasks/DeployAzureResourceGroup/AzureUtilityGTE1.0.ps1 [hunk 1]
```diff
@@ -35,15 +35,17 @@ function Deploy-AzureResourceGroup
           [string]$resourceGroupName,
           [string]$overrideParameters)
 
-    Write-Host "[Azure Resource Manager]Creating resource group deployment with name $resourceGroupName"
+    $deploymentName = [System.IO.Path]::GetFileNameWithoutExtension($csmFile) + '-' + ((Get-Date).ToUniversalTime()).ToString('yyyyMMdd-HHmm')
+
+    Write-Host "[Azure Resource Manager]Creating resource group deployment with name $deploymentName"
```
Comment (pryenugu) [inline Tasks/DeployAzureResourceGroup/AzureUtilityGTE1.0.ps1]: Have $resourceGroupName here instead of the $deploymentName. Additionally it would be even better if you could print something like: "Creating resource group with name: $resourceGroupName and deployment name: $deploymentName".

File: Tasks/DeployAzureResourceGroup/AzureUtilityLTE9.8.ps1 [hunk 2]
```diff
@@ -37,15 +37,18 @@ function Deploy-AzureResourceGroup
           [string]$overrideParameters)
 
     Switch-AzureMode AzureResourceManager
-    Write-Host "[Azure Resource Manager]Creating resource group deployment with name $resourceGroupName"
+
+    $deploymentName = [System.IO.Path]::GetFileNameWithoutExtension($csmFile) + '-' + ((Get-Date).ToUniversalTime()).ToString('yyyyMMdd-HHmm')
+
+    Write-Host "[Azure Resource Manager]Creating resource group deployment with name $deploymentName"
```
Comment (pryenugu) [inline Tasks/DeployAzureResourceGroup/AzureUtilityLTE9.8.ps1]: Have $resourceGroupName here instead of the $deploymentName. Additionally it would be even better if you could print something like: "Creating resource group with name: $resourceGroupName and deployment name: $deploymentName".

**Comment by pryenugu** [comment]
Can you increase the patch version of the task? You should increase it in the task.json of this task and do a gulp (this will cause the task.loc.json as well to be increased in the patch version).

**Comment by pryenugu** [comment]
Also make sure you do a gulp test so that none of the unit tests are breaking.

**Comment by pryenugu** [comment]
The changes by themselves seems good though.

File: Tasks/DeployAzureResourceGroup/AzureUtilityGTE1.0.ps1 (hunk 1 above)
Comment (antmeehan) [inline Tasks/DeployAzureResourceGroup/AzureUtilityGTE1.0.ps1]: The resource group name is displayed earlier (when it is created). This log is for describing the name of the deployment that is about to be created. Make sense?

File: Tasks/DeployAzureResourceGroup/AzureUtilityLTE9.8.ps1 (hunk 2 above)
Comment (antmeehan) [inline Tasks/DeployAzureResourceGroup/AzureUtilityLTE9.8.ps1]: The resource group name is displayed earlier (when it is created). This log is for describing the name of the deployment that is about to be created. Make sense?

**Comment by antmeehan** [comment]
`gulp test` fails when I try:

````
C:\Users\ajmee\git\antmeehan\vso-agent-tasks>gulp test
[16:25:14] Using gulpfile ~\git\antmeehan\vso-agent-tasks\gulpfile.js
[16:25:14] Starting 'cleanTests'...
[16:25:14] Finished 'cleanTests' after 28 ms
[16:25:14] Starting 'compileTests'...
[16:25:15] Compiling TypeScript files using tsc version 1.8.7
[16:25:18] [tsc] > Tests/lib/vsts-task-lib/toolrunner.ts(47,9): error TS17009: 'super' must be called before accessing 'this' in the constructor of a derived class.
[16:25:18] [tsc] > Tests/lib/vsts-task-lib/toolrunner.ts(48,9): error TS17009: 'super' must be called before accessing 'this' in the constructor of a derived class.
[16:25:18] [tsc] > Tests/lib/vsts-task-lib/toolrunner.ts(49,9): error TS17009: 'super' must be called before accessing 'this' in the constructor of a derived class.
[16:25:18] Failed to compile TypeScript: Error: tsc command has exited with code:2

events.js:154
      throw er; // Unhandled 'error' event
      ^
 Error: Failed to compile: tsc command has exited with code:2
````


Any ideas?

**Comment by pryenugu** [comment]
Can you pull the latest code ? it contains a fix for the typescript version 1.8.7

File: Tasks/DeployAzureResourceGroup/AzureUtilityGTE1.0.ps1 (hunk 1 above)
Comment (pryenugu) [inline Tasks/DeployAzureResourceGroup/AzureUtilityGTE1.0.ps1]: Seems good.

File: Tasks/DeployAzureResourceGroup/AzureUtilityLTE9.8.ps1 (hunk 2 above)
Comment (pryenugu) [inline Tasks/DeployAzureResourceGroup/AzureUtilityLTE9.8.ps1]: Seems good.

**Comment by antmeehan** [comment]
Sorry noob GitHub PR question, should I rebase my PR branch? Or merge? (I'm worried the comments will be lost).

**Comment by Ajay-MS** [comment]
@antmeehan No need to rebase you can merge

**Comment by antmeehan** [comment]
Thanks. I've merged and now the tests run. 7 fail, but the did before my change too.

[GulpTestBeforeChange.txt](https://github.com/Microsoft/vso-agent-tasks/files/185637/GulpTestBeforeChange.txt)
[GulpTestAfterChange.txt](https://github.com/Microsoft/vso-agent-tasks/files/185638/GulpTestAfterChange.txt)
